src/plugins/explore/public/application/hooks/editor_hooks/use_language_switch/use_language_switch.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { EditorMode } from '../../../utils/state_management/types';
+import { setEditorMode } from '../../../utils/state_management/slices';
+import { useEditorRef } from '../use_editor_ref';
+
+export interface UseLanguageSwitchReturn {
+  switchEditorMode: (mode: EditorMode) => void;
+}
+
+export const useLanguageSwitch = (): UseLanguageSwitchReturn => {
+  const dispatch = useDispatch();
+  const editorRef = useEditorRef();
+
+  const switchEditorMode = useCallback(
+    (mode: EditorMode) => {
+      dispatch(setEditorMode(mode));
+      // select all
+      const range = editorRef.current?.getModel()?.getFullModelRange();
+      if (range) {
+        setTimeout(() => editorRef.current?.setSelection(range), 300);
+      }
+    },
+    [dispatch, editorRef]
+  );
+
+  return {
+    switchEditorMode,
+  };
```

Comment:
why return an object? can we just have it return `switchEditorMode`? or better yet:

```
return useCallback(...)
```